governance/contracts/ValidatorSet.sol

```diff
@@ -0,0 +1,83 @@
+pragma solidity ^0.5.0;
+
+import "./Ownable.sol";
+
+contract ValidatorSet is Ownable {
+
+  mapping(bytes32 => address) internal _validatorsAddress;
+  string[] public _validatorPubKeys;
+
+  bool public _freeze = false;
+  address public _nextVersion;
+
+  event ValidatorAdded(bytes32 _key, address _address, string _pubKey);
+  event ValidatorRemoved(bytes32 _key, address _address, string _pubKey);
+  event Freeze();
+  event SetNextVersion(address _address);
+
+  function addValidator(string memory _pubKey, address _address) onlyOwner public {
+    require(_freeze == false);
+    require(_address != address(0x0));
+    require(bytes(_pubKey).length == 40);
+
+    bytes32 _vItemKey = calculateValidatorItemKey(_pubKey);
+    require(_validatorsAddress[_vItemKey] == address(0x0));
+    
+    _validatorsAddress[_vItemKey] = _address;
+    _validatorPubKeys.push(_pubKey);
+
+    emit ValidatorAdded(_vItemKey, _address, _pubKey);
+  }
+  
+  function removeValidator(string memory _pubKey, address _address) onlyOwner public {
+    require(_freeze == false);
+    require(_address != address(0x0));
+    require(bytes(_pubKey).length == 40);
+
+    bytes32 _vItemKey = calculateValidatorItemKey(_pubKey);
+    require(_validatorsAddress[_vItemKey] == _address);
+
+    _validatorsAddress[_vItemKey] = address(0x0);
+    bool _foundDeletedItem = false;
+    for (uint i = 0; i < _validatorPubKeys.length-1; i++) {
+      if (calculateValidatorItemKey(_validatorPubKeys[i]) == _vItemKey) {
+        _foundDeletedItem = true;
+      }
+      if (_foundDeletedItem) {
+        _validatorPubKeys[i] = _validatorPubKeys[i+1];
+      }
+    }
+
+    delete _validatorPubKeys[uint(_validatorPubKeys.length-1)];
+    _validatorPubKeys.length--;
+    emit ValidatorRemoved(_vItemKey, _address, _pubKey);
+  }
+  
+  function validatorAddress(string memory _pubKey) public view returns (address) {
+    bytes32 _vItemKey = calculateValidatorItemKey(_pubKey);
+    return _validatorsAddress[_vItemKey];
+  }
+  
+  function validatorPubKey(uint index) public view returns (string memory) {
+    return _validatorPubKeys[index];
+  }
+
+  function setFreezeStatus(bool _value) onlyOwner public {
+    _freeze = _value;
+    emit Freeze();
+  }
+  
+  
+  function calculateValidatorItemKey(string memory _pubKey) pure internal returns (bytes32) {
+    return sha256(bytes(_pubKey));
+  }
+
+  function validatorSetSize() public view returns (uint) {
```

Comment:
Keep in mind this is a matter of personal preferences. I am more inclined to keep the  descriptive  method names even though it might be redundant for some people.

Reply:
Yea I fully hear you and normally I am a big fan of descriptive methods as you know and funny enough you are normally against them haha. The thing is, this is a standard Set and I have never seen one with descriptive methods:

https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html
https://www.php.net/manual/en/class.ds-set.php

It has all the methods: "add", "remove", "size"... would be really more inclined in calling them in a standard short form.

`ValidatorSet->validatorSetSize()` seems really unnecessary
`ValidatorSet->size()` niceee :)